src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -675,83 +444,68 @@ ${if (createTable) {
-          input_relations: [${
-            (stmts flatMap ss.relationNamesUsedByStatement distinct
-            ) mkString("\n            ", "\n            ", "\n          ")}]
-      }
-    """
-    List(extractor)
+}),
+      "output_relation" -> stmts(0).headName,
+      "style" -> "cmd_extractor",
+      "input_relations" -> (stmts flatMap relationNamesUsedByStatement distinct)
+    ))
   }
 
-  def compileFunctionCallRules(stmts: List[FunctionCallRule], ss: CompilationState): CompiledBlocks = {
-    var extractors = new ListBuffer[String]()
-    for (stmt <- stmts) {
-
-      val function = ss.resolveFunctionName(stmt.function)
-      val udfDetails = (function.implementations collectFirst {
-        case impl: RowWiseLineHandler =>
-          s"""udf: $${APP_HOME}\"/${StringEscapeUtils.escapeJava(impl.command)}\"
-          style: \"${impl.style}_extractor\" """
-      })
+  def compile(stmt: FunctionCallRule): CompiledBlocks = {
+    val stmts = statementsByRepresentative getOrElse(stmt, List.empty) collect { case s: FunctionCallRule => s }
+    if (stmts isEmpty) return List.empty
+
+    stmts map { case stmt =>
+      val function = functionDeclarationByName(stmt.function)
+      val udfDetails = function.implementations collectFirst {
+        case impl: RowWiseLineHandler => List(
+            "udf" -> QuotedString(impl.command),
+            "style" -> s"${impl.style}_extractor"
+          )
+      }
 
       if (udfDetails.isEmpty)
-        ss.error(s"Cannot find compilable implementation for function ${stmt.function} among:\n  "
+        sys.error(s"Cannot find compilable implementation for function ${stmt.function} among:\n  "
           + (function.implementations mkString "\n  "))
 
-      val blockName = ss.resolveExtractorBlockName(stmt)
+      val blockName = resolveExtractorBlockName(stmt)
       val parallelism = stmt.parallelism getOrElse("${PARALLELISM}")
-      val extractor = s"""
-        deepdive.extraction.extractors.${blockName} {
-          input: \"\"\" ${new QueryCompiler(stmt.q, ss).generateSQL(TableAlias)}
-          \"\"\"
-          output_relation: \"${stmt.output}\"
-          ${udfDetails.get}
-          ${ss.generateDependenciesOfCompiledBlockFor(List(stmt))}
-          input_relations: [${
-            (List(stmt) flatMap ss.relationNamesUsedByStatement distinct
-            ) mkString("\n            ", "\n            ", "\n          ")}]
-          input_batch_size: $${INPUT_BATCH_SIZE}
-          parallelism: ${parallelism}
-        }
-      """
-      extractors += extractor
+      s"deepdive.extraction.extractors.${blockName}" -> (Map(
+        "input" -> QuotedString(new QueryCompiler(stmt.q).generateSQL(TableAlias)),
+        "output_relation" -> stmt.output,
+        "input_relations" -> (List(stmt) flatMap relationNamesUsedByStatement distinct),
+        "input_batch_size" -> "${INPUT_BATCH_SIZE}",
+        "parallelism" -> parallelism
+        ) ++ udfDetails.get)
```

Comment:
Just to double check. Dependency is dropped here because it will be handled in deepdive. Is that correct?